core/go/internal/componentmgr/manager.go

```diff
@@ -315,6 +315,9 @@ func (cm *componentManager) startBlockIndexer() (err error) {
 		_, err = cm.blockIndexer.GetBlockListenerHeight(cm.bgCtx)
 		err = cm.wrapIfErr(err, msgs.MsgComponentBlockIndexerStartError)
 	}
+	if err == nil {
+		err = cm.txManager.LoadBlockchainEventListeners()
```

Comment:
Looking into the code I see why you made this a whole separate lifecycle (rather than just make these new returned `internalEventStreams`).

Overall this function is a bit clunky and ❄️ in the component initialization - but that was true before your addition.